src/iperf3test_cli.py

```diff
@@ -0,0 +1,161 @@
+        server_args = ""
+
+        server_thread = threading.Thread(target=start_server, args=(self.server, server_args))
+        server_thread.start()
+
+        client_thread = threading.Thread(target=start_client, args=(self.client, self.server, client_args))
+        client_thread.start()
+
+    def run_packet_count_test(self):
+        run_duration = 20
+        client_args = f"--parallel 128 --time {run_duration} --set-mss 89"
+        server_args = ""
+
+        print(f"Testing pps for {self.client['public_ip']} > {self.server['public_ip']}")
+
+        server_thread = threading.Thread(target=start_server, args=(self.server, server_args, True))
+        server_thread.start()
+
+        client_thread = threading.Thread(target=start_client, args=(self.client, self.server, client_args, True))
+        client_thread.start()
+
+        server_pps_thread = threading.Thread(target=start_pps, args=(self.server, run_duration))
+        server_pps_thread.start()
+
+        server_thread.join()
+        client_thread.join()
+        server_pps_thread.join()
+
+        print(f"Testing pps for {self.client['public_ip']} < {self.server['public_ip']}")
+
+        server_thread = threading.Thread(target=start_server, args=(self.client, server_args, True))
+        server_thread.start()
+
+        client_thread = threading.Thread(target=start_client, args=(self.server, self.client, client_args, True))
+        client_thread.start()
+
+        server_pps_thread = threading.Thread(target=start_pps, args=(self.server, run_duration))
+        server_pps_thread.start()
+
+
+def start_client(client, server, args, silent=False):
+    sleep(2)
+    ssh_client = Ssh(
+        client["public_ip"],
+        client["ssh_user"],
+        client["ssh_options"],
+        log_enabled=False,
+        use_control_socket=False)
+    ssh_client.connect()
+    server_ip = server["private_ip"]
+    iperf_cmd = f"iperf3 -c {server_ip} -p 3000 {args} --connect-timeout 30000"
+    print(iperf_cmd)
+    ssh_client.exec(iperf_cmd, silent)
+
+
+def start_server(server, args, silent=False):
+    ssh_server = Ssh(
+        server["public_ip"],
+        server["ssh_user"],
+        server["ssh_options"],
+        silent_seconds=0,
+        log_enabled=False,
+        use_control_socket=False)
+    ssh_server.connect()
+    iperf_cmd = f"iperf3 --one-off -s -p 3000 --pidfile iperf3.pid {args}"
+    print(iperf_cmd)
+    ssh_server.exec(f"""
+        set -e
+        if [ -e "iperf3.pid" ]; then
+            (kill -9 $(cat iperf3.pid)) >/dev/null 2>&1
+            rm iperf3.pid
+        fi
+        {iperf_cmd}
+        echo iperf3 server done
+        """, silent)
+
+
+def start_pps(server, time):
+    ssh_server = Ssh(
+        server["public_ip"],
+        server["ssh_user"],
+        server["ssh_options"],
+        log_enabled=False,
+        use_control_socket=False)
+    ssh_server.connect()
+    cmd = f"hazelcast-simulator/bin/pps ens5 {time}"
```

Comment:
Yes. This needs to be fixed.

Reply:
Either disable this command or automatically configure the interface correctly. In /sys/class/net you can see the interfaces. If there are 2, then you can remove localhost. Otherwise throw an error.

I don't expect this tool will be valuable since we want to collect pps automatically.